cmd/cluster-agent/app/app.go

```diff
@@ -244,10 +245,19 @@ func start(cmd *cobra.Command, args []string) error {
 	sc := clusteragent.ServerContext{
 		ClusterCheckHandler: clusterCheckHandler,
 	}
+
 	if err = api.StartServer(sc); err != nil {
 		return log.Errorf("Error while starting agent API, exiting: %v", err)
```

Comment:
Ah, ok. I see where this mistake came from :) Here also the returned error will be wrong (and likely `nil`). Same bug.

Reply:
Fixed in all related places.